### PR TITLE
[FLINK-33860] Implement restore tests for WindowTableFunction node

### PR DESCRIPTION
**Pull request by bvarghese1**
## What is the purpose of the change

*Add restore tests for WindowTableFunction node*

## Verifying this change

This change added tests and can be verified as follows:

- Added restore tests for WindowTableFunction node which verifies the generated compiled plan with the saved compiled plan

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/WindowTableFunctionJsonPlanTest.java [hunk 1]
```diff
@@ -1,210 +0,0 @@
...
-                        + "  window_start,\n"
-                        + "  window_end,\n"
-                        + "  a,\n"
-                        + "  b,\n"
-                        + "  c\n"
-                        + "FROM TABLE(TUMBLE(TABLE MyTable, DESCRIPTOR(proctime), INTERVAL '15' MINUTE))");
-    }
-
-    @Test
-    void testFollowedByWindowJoin() {
```
Comment (bvarghese1) [inline flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/WindowTableFunctionJsonPlanTest.java:110]: This test is covered as part of the WindowJoin restore tests

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/WindowTableFunctionJsonPlanTest.java [hunk 2]
```diff
@@ -1,210 +0,0 @@
...
-                        + "    b,\n"
-                        + "    c\n"
-                        + "  FROM TABLE(TUMBLE(TABLE MyTable2, DESCRIPTOR(rowtime), INTERVAL '15' MINUTE))\n"
-                        + "  WHERE b > 10\n"
-                        + ") R\n"
-                        + "ON L.window_start = R.window_start AND L.window_end = R.window_end AND L.a = R.a");
-    }
-
-    @Test
-    void testFollowedByWindowRank() {
```
Comment (bvarghese1) [inline flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/WindowTableFunctionJsonPlanTest.java:158]: This test will be covered as part of the WindowRank restore tests

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/WindowTableFunctionJsonPlanTest.java [hunk 3]
```diff
@@ -1,210 +0,0 @@
...
-                        + "FROM (\n"
-                        + "  SELECT\n"
-                        + "    *,\n"
-                        + "   ROW_NUMBER() OVER(PARTITION BY a, window_start, window_end ORDER BY b DESC) as rownum\n"
-                        + "  FROM TABLE(TUMBLE(TABLE MyTable, DESCRIPTOR(rowtime), INTERVAL '15' MINUTE)))\n"
-                        + "WHERE rownum <= 3");
-    }
-
-    @Test
-    void testFollowedByWindowDeduplicate() {
```
Comment (bvarghese1) [inline flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/WindowTableFunctionJsonPlanTest.java:185]: This test is covered as part of the WindowDeduplicate restore tests

**Review by dawidwys — APPROVED** [review]
LGTM % CI failure

**Pull request merged**